packages/aws-rfdk/lib/lambdas/nodejs/lib/configure-spot-event-plugin/test/spot-event-plugin-client.test.ts

```diff
@@ -238,4 +250,111 @@ describe('SpotEventPluginClient', () => {
     // THEN
     await expect(promise).rejects.toEqual(statusMessage);
   });
+
+  test('successful getGroupPoolCollection for groups', async () => {
+    // WHEN
+    // eslint-disable-next-line dot-notation
+    spotEventPluginClient['deadlineClient'].GetRequest = jest.fn().mockResolvedValue(successfulPoolResponse);
+    const result = await spotEventPluginClient.getGroupPoolCollection(true);
+
+    // THEN
+    expect(result).toEqual(deadlinePools);
```

Comment:
nit: variable name is misleading. While they are both just string arrays, I think it's worth creating a separate `deadlineGroups` variable so that our tests are clear about what they are testing.